### ISSUE TITLE
Image slide javascripts has no effect on my_module.js html

### ISSUE THREAD

**Issue by afcdotcom** [signalpoint/DrupalGap#388]
Hi,

I tried a few image slide javascripts but with the same results: with the html in index.html everything works fine, but with the html inside my_module.js nothing happens. I used something like this:

content['slide'] = {
        markup: "<div>My html here...</div>"
}

Can you help me please? What am I doing wrong?

**Comment by signalpoint** [comment]
@afcdotcom Please provide much more context and code you are using, I'm not sure how to help. Other than to try a pageshow event instead to initiate any JavaScript funkiness.

http://www.drupalgap.org/node/154

**Comment by afcdotcom** [comment]
@signalpoint How can I post code here because it is messed up after posting the message?

**Comment by signalpoint** [comment]
@afcdotcom Check out the "Parsed as Markdown" link located just above the comment window: https://guides.github.com/features/mastering-markdown/

Essentially you wrap your code in 3 "slanted single quotes" (as you can see I have no idea what that key is called, but it is located typically to the left of the 1 key (just above your keyboard's letters).

**Comment by afcdotcom** [comment]
@signalpoint i think i got it, thanks for explanations. So let's take an example:

In index.html I have something like this:

Head:

```
<script>
$(document).ready(function(){
  $("p").on("swipeleft",function(){
    alert("You swiped left!");
  });                       
});
</script>
```

Body:

```
<p>Swipe me in the left direction!</p>
```

In this case it works, but if I am doing that in my_module.js instead of index.html's body it doesn't:

my_module.js:

```
function my_module_menu() {
    var items = {
        homepage: {
            title: 'my_title',
            page_callback: 'my_module_homepage',
        }
    }
    return items;
}

function my_module_homepage() {
    var content = {};
    content['paragrah_swipe'] = {
       markup: "<p>Swipe me in the left direction!</p>"
    }
    return content;
}
```

Am I missing something?

**Comment by signalpoint** [comment]
@afcdotcom I'm guessing it has to do with the `div` wrappers that are placed around your page by DrupalGap. Try changing your swipeleft css identifier for jQuery to use the `div` `id` on that page. Use your HTML inspector to find the id (I think it will be `homepage`).

**Comment by afcdotcom** [comment]
@signalpoint yes, 'homepage' is the id of my homepage. I tried the code below with no results. It's like it doesn't recognise this kind of script inside drupalgap i guess.

```
<script>
            $(document).ready(function(){
                $("div#homepage").on("swipeleft",function(){
                    alert("You swiped left!");
                });                       
            });
</script>
```

**Comment by afcdotcom** [comment]
@signalpoint I got it with my easy example. The solution is somehing like this: 

```
<script>
            $(document).on("pagecreate","#homepage",function(){
                $("p.swipe").on("swipeleft",function(){
                    alert("You swiped left!");
                });                       
            });
</script>
```

Now let's see if I can make the image slider work with swipe gestures. Please don't close the thread yet, maybe I will still have some related questions until the end of it. Thank you.

**Comment by signalpoint** [comment]
@afcdotcom Excellent, so you are able to detect a swipe on a page in DrupalGap? If so, it'd be great if you could make a page on drupalgap.org that explains how to do it, thanks!
